Linkpoint/src/main/app/com/linkpoint/ui/SettingsActivity.kt

```
@@ -54,57 +83,287 @@ class SettingsActivity : AppCompatActivity() {
+                    val stats = cacheManager.getCacheStats()
+                    findPreference<Preference>("cache_status")?.apply {
+                        val statusText = buildString {
+                            append("Using ${stats.getFormattedTotalSize()} of ${stats.getFormattedMaxSize()}")
+                            append(" (${stats.usagePercent}%)")
+                            if (stats.isLowSpace) {
+                                append("\n⚠️ Device storage is low!")
+                            }
+                        }
+                        summary = statusText
+                    }
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)